statsmodels/tools/rng_qrng.py

```diff
@@ -0,0 +1,43 @@
+import numbers
+from collections import Sequence
+
+import numpy as np
+import scipy.stats as stats
+
+
+def check_random_state(seed=None):
+    """Turn `seed` into a random number generator.
+
+    Parameters
+    ----------
+    seed : {None, int, array_like[ints], `numpy.random.Generator`,
+            `numpy.random.RandomState`, `scipy.stats.qmc.QMCEngine`}, optional
+
+        If `seed` is None fresh, unpredictable entropy will be pulled
+        from the OS and `numpy.random.Generator` is used.
+        If `seed` is an int or ``array_like[ints]``, a new ``Generator``
+        instance is used, seeded with `seed`.
+        If `seed` is already a ``Generator``, ``RandomState`` or
+        `scipy.stats.qmc.QMCEngine` instance then
+        that instance is used.
+
+        `scipy.stats.qmc.QMCEngine` requires SciPy >=1.7. It also means
+        that the generator only have the method ``random``.
+
+    Returns
+    -------
+    seed : {`numpy.random.Generator`, `numpy.random.RandomState`,
+            `scipy.stats.qmc.QMCEngine`}
+
+        Random number generator.
+
+    """
+    if seed is None or isinstance(seed, (numbers.Integral, np.integer, Sequence)):
```

Comment:
I think your idea to invert - specific instance first than pass remainder to `default_rng()` and let NumPy raise.

Reply:
It is quite a simplification. Hence there is just a need to check for `RandomState` and `QMCEngine` as `Generator` can be passed to `default_rng`